smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -187,12 +203,169 @@ private List<ParameterObject> createPathParameters(Context<T> context, Operation
+                            type == MessageType.REQUEST ? example.getInput() : example.getOutput());
+                    String name = operationOrError.getId().getName() + "_example" + uniqueNum++;
+                    // this if condition is needed to avoid errors when converting examples of response.
+                    if (!example.getError().isPresent() || type == MessageType.REQUEST) {
+                        examples.put(name, ExampleObject.builder()
+                                .summary(example.getTitle())
+                                .description(example.getDocumentation().orElse(""))
+                                .value(transformSmithyValueToProtocolValue(values))
+                                .build());
+                    }
+                }
+            }
+            return examples;
+        }
+    }
+
+    private Map<String, ExampleObject> createErrorBodyExamples(Shape error, List<HttpBinding> bindings,
+                                                               OperationShape operation) {
+        Map<String, ExampleObject> examples = new TreeMap<>();
+        // unique numbering for unique example names in OpenAPI.
+        int uniqueNum = 1;
+        Optional<ExamplesTrait> examplesTrait = operation.getTrait(ExamplesTrait.class);
+        if (examplesTrait.isPresent()) {
+            for (ExamplesTrait.Example example : examplesTrait.get().getExamples()) {
+                // this has to be checked because an operation can have more than one error linked to it.
+                if (example.getError().isPresent()
+                        && example.getError().get().getShapeId() == error.toShapeId()) {
+                    // get members included in bindings
+                    ObjectNode values = filter(bindings, example.getError().get().getContent());
+                    String name = operation.getId().getName() + "_example" + uniqueNum;
+                    examples.put(name,
+                            ExampleObject.builder()
+                                    .summary(example.getTitle())
+                                    .description(example.getDocumentation().orElse(""))
+                                    .value(transformSmithyValueToProtocolValue(values))
+                                    .build());
+                }
+                uniqueNum++;
+            }
+        }
+        return examples;
+    }
+
+    /*
+     * Returns a modified copy of [inputOrOutput] with just the members bound to a HttpBinding trait in [bindings].
+     */
+    private ObjectNode filter(List<HttpBinding> bindings, ObjectNode inputOrOutput) {
+        ObjectNode values = inputOrOutput;
+        ObjectNode tmp = inputOrOutput.toBuilder().build();
+
+        for (HttpBinding binding : bindings) {
+            tmp = tmp.withoutMember(binding.getMemberName());
+        }
+        for (StringNode key : tmp.getMembers().keySet()) {
+            values = values.withoutMember(key.toString());
+        }
```

Comment:
Overall, performance here isn't critical, but this code inefficiently creates excessive objects that get thrown away immediately. Another approach would be to use an `ObjectNode.Builder` and only adding members that are found from `bindings[N].getMemberName()` (psuedo-code). A bigger benefit to me would be read clarity.

Reply:
Changed to reduce inefficiency & improve readability, to following:

```
private ObjectNode getMembersWithHttpBindingTrait(List<HttpBinding> bindings, ObjectNode inputOrOutput) {
        ObjectNode.Builder values = ObjectNode.builder();

        Set<String> memberNamesWithHttpBinding = bindings.stream()
                                                    .map(HttpBinding::getMemberName)
                                                    .collect(Collectors.toSet());

        for (Map.Entry<StringNode, Node> entry : inputOrOutput.getMembers().entrySet()) {
            if (memberNamesWithHttpBinding.contains(entry.getKey().toString())) {
                values.withMember(entry.getKey(), entry.getValue());
            }
        }

        return values.build();
    }
```